todo_app/tests/test_integration.py

```diff
@@ -0,0 +1,157 @@
+import pytest, os
+from dotenv import find_dotenv, load_dotenv
+import todo_app.app as app
+from unittest.mock import patch, Mock
+
+@pytest.fixture
+def client():
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+    test_app = app.create_app()
+    with test_app.test_client() as client: 
+        yield client
+
+@patch('requests.get')
+def test_index_page(mock_get, client):
+    mock_get.side_effect = mock_get_cards
+    response = client.get('/')
+    print(response.data.decode())
+    assert response.status_code == 200
+
+def mock_get_cards(url, params=None):
+
+    if url == f'https://trello.com/1/lists/todo_list_id/cards?key=test_key&token=test_token': 
```

Comment:
This also assumes that the url will be passed through to the mocked function with query parameters already baked in, but your code passes those through in the params dictionary

Reply:
yep, changed it as needed, thanks.